plugins/updater/src/updater.rs

```diff
@@ -534,118 +541,50 @@ impl Update {
-                let mut current_exe_arg = std::ffi::OsString::new();
-                current_exe_arg.push("\"");
-                current_exe_arg.push(current_exe()?);
-                current_exe_arg.push("\"");
-
-                let mut msi_path = std::ffi::OsString::new();
-                msi_path.push("\"\"\"");
-                msi_path.push(&found_path);
-                msi_path.push("\"\"\"");
-
-                let installer_args = [
+                installer_args.extend(
                     install_mode
                         .msiexec_args()
                         .iter()
                         .map(OsStr::new)
                         .collect::<Vec<_>>(),
-                    self.installer_args
-                        .iter()
-                        .map(|a| a.as_os_str())
-                        .collect::<Vec<_>>(),
-                ]
-                .concat();
-
-                // run the installer and relaunch the application
-                let powershell_install_res = Command::new(powershell_path)
-                    .args(["-NoProfile", "-WindowStyle", "Hidden"])
-                    .args([
-                        "Start-Process",
-                        "-Wait",
-                        "-FilePath",
-                        "$Env:SYSTEMROOT\\System32\\msiexec.exe",
-                        "-ArgumentList",
-                    ])
-                    .arg("/i,")
-                    .arg(&msi_path)
-                    .arg(format!(
-                        ", {}, /promptrestart;",
-                        installer_args.join(OsStr::new(", ")).to_string_lossy()
-                    ))
-                    .arg("Start-Process")
-                    .arg(current_exe_arg)
-                    .spawn();
-                if powershell_install_res.is_err() {
-                    // fallback to running msiexec directly - relaunch won't be available
-                    // we use this here in case powershell fails in an older machine somehow
-                    let msiexec_path = system_root.as_ref().map_or_else(
-                        |_| "msiexec.exe".to_string(),
-                        |p| format!("{p}\\System32\\msiexec.exe"),
-                    );
-                    let _ = Command::new(msiexec_path)
-                        .arg("/i")
-                        .arg(msi_path)
-                        .args(installer_args)
-                        .arg("/promptrestart")
-                        .spawn();
-                }
-
-                std::process::exit(0);
+                );
+                installer_args.push(OsStr::new("/promptrestart"));
+            } else {
+                continue;
+            }
+            let file = HSTRING::from(found_path.as_os_str());
+            let parameters = HSTRING::from(installer_args.join(OsStr::new(" ")));
+            let ret =
+                unsafe { ShellExecuteW(None, w!("runas"), &file, &parameters, None, SW_SHOW) };
```

Comment:
should probably use `open` instead of `runas`

Reply:
It was `runas` in the powershell version, just migrated as is, I'm not sure should we run it elevated by default or not

Reply:
we were using `Start-Process` without `-Verb` which defaults to `open` iirc